### PR TITLE
fix: count error & totalPage error (#1373)

### PR DESCRIPTION
**Pull request by LetFu**
Signed-off-by: LetFu <letfu@outlook.com>

## Description

Reset offset before calculating count;
Fix the totalpage calculation error.

## Related Issue

#1373

## Motivation and Context

I looked at the code and I found that the error should be here:
https://github.com/dragonflyoss/Dragonfly2/blob/main/manager/service/job.go#L180-L184
```go
if err := s.db.WithContext(ctx).Scopes(model.Paginate(q.Page, q.PerPage)).Where(&model.Job{
    Type:   q.Type,
    State:  q.State,
    UserID: q.UserID,
}).Find(&jobs).Count(&count).Error; err != nil {
```
When page is not 1, the offset calculated by the following code will not be 0
https://github.com/dragonflyoss/Dragonfly2/blob/main/manager/model/models.go#L40
```go
offset := (page - 1) * perPage
```
At this time, the SQL statement generated by the above SQL code is as follows:
```sql
{"elapsed": "2.223839ms", "rows": 2, "sql": "SELECT * FROM `job` WHERE `job`.`is_del` = 0 LIMIT 50 OFFSET 50"}
{"elapsed": "872.777µs", "rows": 0, "sql": "SELECT count(*) FROM `job` WHERE `job`.`is_del` = 0 LIMIT 50 OFFSET 50"}  # result 0
```

Also, I found that the `totalPage` is wrongly calculated here, resulting in the wrong `header.link` being returned.
https://github.com/dragonflyoss/Dragonfly2/blob/main/manager/handlers/handlers.go#L50

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by LetFu** [comment]
> Please rebase main branch.

Ok, I've rebase, please review, thx.

File: manager/handlers/handlers.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func (h *Handlers) setPaginationDefault(page, perPage *int) {
 }
 
 func (h *Handlers) setPaginationLinkHeader(ctx *gin.Context, page, perPage, totalCount int) {
-	totalPage := totalCount / perPage
+	totalPage := totalCount/perPage + 1
 	if totalPage == 0 {
```
Comment (gaius-qi) [inline manager/handlers/handlers.go:51]: This part of the code can be deleted.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: manager/handlers/handlers.go (hunk 1 above)
Comment (LetFu) [reply manager/handlers/handlers.go]: Ok, it's deleted

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by hyy0322 — APPROVED** [review]
LGTM

**Pull request merged**